### ISSUE TITLE
IAM roles not created consistently

### ISSUE THREAD

**Issue by eeaton** [terraform-google-modules/terraform-example-foundation#1128]
### TL;DR

Only some groups automatically have the recommended IAM roles defined. Some groups are setup in the initial section but never used in IAM policy, even though documentation at [groups for access control](https://cloud.google.com/architecture/security-foundations/authentication-authorization#groups_for_access_control) recommends it.

It's confusing that some groups recommended for operational functions are optional (`gcp_security_reviewer`) and some are required (`monitoring_workspace_users`). I expected that the infra should still deploy with or without the group.

After investigating, I see the inconsistent behavior for "required" is because TF code explicitly creates the [IAM role for the monitoring group](https://github.com/terraform-google-modules/terraform-example-foundation/blob/dd1952d171cd369f2de534f28ed678dca71acb12/2-environments/modules/env_baseline/iam.tf#L21), but it [does not create any IAM role for the security group](https://github.com/search?q=repo%3Aterraform-google-modules%2Fterraform-example-foundation%20gcp_security_reviewer&type=code).

### Expected behavior

Each group defined in both the `required_groups` and `optional_groups` should have IAM roles applied, as defined in the [groups for access control](https://cloud.google.com/architecture/security-foundations/authentication-authorization#groups_for_access_control) guidance.

### Observed behavior

Only those groups in the `required_groups` have IAM roles applied.

### Terraform Configuration

```hcl
n/a
```

### Terraform Version

```sh
n/a
```
